FhirToDataLake/native/parquet/cpp/test/SchemaManagerTests.cpp

```diff
@@ -48,22 +48,37 @@ TEST (SchemaTest, Valid)
     EXPECT_EQ(1, schema->num_fields());
 }
 
-TEST (SchemaTest, Empty)
+TEST (SchemaTest, AddAndGetEmptySchemaContent)
 {
     SchemaManager schemaManager;
+    string schemaKey = "Empty";
     string empty = "";
-    int status = schemaManager.AddSchema("Empty", empty);
-    auto emptySchema = schemaManager.GetSchema("Empty");
+    int status = schemaManager.AddSchema(schemaKey, empty);
     EXPECT_EQ(11001, status);
+
+    auto emptySchema = schemaManager.GetSchema(schemaKey);
     EXPECT_TRUE(emptySchema == nullptr);
 }
 
-TEST (SchemaTest, Invalid)
+TEST (SchemaTest,  AddAndGetNoneJsonSchemaContent)
 {
     SchemaManager schemaManager;
+    string schemaKey = "Broken";
     string broken = "abc";
-    int status = schemaManager.AddSchema("Broken", broken);
-    auto brokenSchema = schemaManager.GetSchema("Broken");
+    int status = schemaManager.AddSchema(schemaKey, broken);
     EXPECT_EQ(11001, status);
+
+    auto brokenSchema = schemaManager.GetSchema(schemaKey);
     EXPECT_TRUE(brokenSchema == nullptr);
-}
\ No newline at end of file
+}
+
+TEST (SchemaTest,  AddAndGetEmptyOrWhiteSpaceSchemaKey)
+{
+    SchemaManager schemaManager;
+    string mockSchema = "{\"Name\": \"Organization\", \"NodePaths\": [\"Organization\"], \"SubNodes\": { \"id\": {\"Name\":\"id\", \"Depth\": 1, \"Type\": \"id\", \"IsLeaf\": true, \"IsRepeated\": false}}, \"Type\": \"Organization\", \"IsRepeated\": false}";
```

Comment:
Can we also add tests for schema that has some required field missing like id, nodepath, isrepeated, isleaf?

Reply:
Added more tests in Native CSharp project, here is no "required" field, the field can be missed as long as it matched with input data.

The default property type is string, default IsRepeated and IsLeaf is false. 

I added a simplest schema test file, only contains all used sections to correctly convert a data.